src/target/llvm/codegen_llvm.cc

```diff
@@ -906,7 +954,7 @@ DEFINE_CODEGEN_BINARY_OP(Mul);
   llvm::Value* CodeGenLLVM::Create##Op(DataType t, llvm::Value* a, llvm::Value* b) { \
     if (t.is_int()) {                                                                \
       return builder_->CreateICmpS##Op(a, b);                                        \
-    } else if (t.is_uint()) {                                                        \
+    } else if (t.is_uint() || t.is_bfloat()) {                                       \
```

Comment:
Isn't comparing bfloat16 this way risky?

Reply:
FP32/FP64 comparasion are also bit-wise in my understanding.